api/src/main/java/org/openmrs/module/fhir/api/impl/PersonServiceImpl.java

```diff
@@ -114,4 +116,23 @@ public Person updateFHIRPerson(Person thePerson, String theId) {
 			return createFHIRPerson(thePerson);
 		}
 	}
+
+	/**
+	 * @see org.openmrs.module.fhir.api.PersonService#retirePerson(String)
+	 */
+	@Override
+	public void retirePerson(String id) throws ResourceNotFoundException, NotModifiedException {
+		org.openmrs.Person person = Context.getPersonService().getPersonByUuid(id);
+		if (person == null) throw new ResourceNotFoundException(String.format("Person with id '%s' not found", id));
+		if (person.isPersonVoided()) return;
+		try {
+			person.setVoided(true);
```

Comment:
Setting only person to be void with reason should be fine. Rest underline API invocation should add :).